### PR TITLE
MINOR: ConfigDef `parseType` exception message updated.

### PR DESCRIPTION
**Pull request by kamalcph**
- When Kafka configurations are provided programmatically, for the below configuration incorrect error message gets printed.
    * configuration: kafkaProps.put("zookeeper.session.timeout.ms", 60_000L); - expects INTEGER value.
    * Error Msg: "Invalid value 60000 for configuration zookeeper.session.timeout.ms : Expected value to be an number."
    * Long is also a number which misleading the error message.
- Minor code cleanup and unwanted local variables are removed.
- Java doc dangling imports are updated.

### CODE REVIEW

**Comment by kamalcph** [comment]
Is it not allowed in check-style to add imports for Java-doc ?

**Comment by ijuma** [comment]
Thanks for the PR. The clean-ups seem completely unrelated to the change here, so it would make more sense to submit a separate PR for them.

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java [hunk 1]
```diff
@@ -73,11 +73,8 @@ public int partition(String topic, Object key, byte[] keyBytes, Object value, by
     private int nextValue(String topic) {
         AtomicInteger counter = topicCounterMap.get(topic);
         if (null == counter) {
-            counter = new AtomicInteger(new Random().nextInt());
-            AtomicInteger currentCounter = topicCounterMap.putIfAbsent(topic, counter);
-            if (currentCounter != null) {
-                counter = currentCounter;
-            }
+            topicCounterMap.putIfAbsent(topic, new AtomicInteger(new Random().nextInt()));
+            counter = topicCounterMap.get(topic);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java]: This adds an additional `get` when compared to the previous solution.

File: clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java [hunk 2]
```diff
@@ -134,8 +134,8 @@ private void update(Map<TopicPartition, S> partitionToState) {
         private final S value;
 
         public PartitionState(TopicPartition topicPartition, S state) {
-            this.topicPartition = Objects.requireNonNull(topicPartition);
-            this.value = Objects.requireNonNull(state);
+            this.topicPartition = Utils.notNull(topicPartition);
+            this.value = Utils.notNull(state);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java]: Not sure why this is being changed.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java [hunk 3]
```diff
@@ -50,10 +50,9 @@
         int maxKey = -1;
         for (ApiKeys key : ApiKeys.values())
             maxKey = Math.max(maxKey, key.id);
-        ApiKeys[] idToType = new ApiKeys[maxKey + 1];
+        ID_TO_TYPE = new ApiKeys[maxKey + 1];
         for (ApiKeys key : ApiKeys.values())
-            idToType[key.id] = key;
-        ID_TO_TYPE = idToType;
+            ID_TO_TYPE[key.id] = key;
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: The idea is to not mutate the map after assignment, so the previous code was fine.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java [hunk 4]
```diff
@@ -64,6 +63,9 @@
     public final String name;
 
     ApiKeys(int id, String name) {
+        if (id < MIN_API_KEY) {
+            throw new IllegalArgumentException("APIKeys id should not be lesser than " + MIN_API_KEY);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: `lesser` -> `less`.

**Review by ijuma — COMMENTED** [review]
Left a few comments as well.

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java (hunk 1 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java]: Agreed. Will revert this change.

File: clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java (hunk 2 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java]: Ok. Will revert this one too.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java (hunk 3 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: I'm not aware of the idea. Will revert this one.

File: clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java (hunk 4 above)
Comment (kamalcph) [reply clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java]: Ok.

**Comment by kamalcph** [comment]
@ijuma 

Thanks for the review. I'll close this one and open a separate request for this PR and the code cleanups.

**Pull request closed without merging**